va/caa.go

```diff
@@ -624,17 +654,39 @@ var validationMethodRegexp = regexp.MustCompile(`^[[:alnum:]-]+$`)
 
 // caaValidationMethodMatches checks that the validationmethods CAA parameter,
 // if present, contains the exact name of the ACME validation method used to
-// validate this domain.
-// See RFC 8657 Section 4: https://www.rfc-editor.org/rfc/rfc8657.html#section-4
-func caaValidationMethodMatches(caaParams map[string]string, method core.AcmeChallenge) bool {
-	commaSeparatedMethods, ok := caaParams["validationmethods"]
-	if !ok {
+// validate this domain. We accept only a single "validationmethods" parameter
+// and will fail if multiple are found in the CAA RR, even if all tag-value
+// pairs would be valid. See RFC 8657 Section 4:
+// https://www.rfc-editor.org/rfc/rfc8657.html#section-4.
+func caaValidationMethodMatches(caaParams []caaParameter, method core.AcmeChallenge) bool {
+	var validationMethods string
+	var found bool
+	for _, param := range caaParams {
+		if param.tag == "validationmethods" {
+			if found {
+				// RFC 8657 does not define what behavior to take when multiple
```

Comment:
I'm not sure I agree with this approach. The reason multiple accounturi parameters is unsatisfiable is that they are by definition unique, so the intersection of "uriA" and "uriB" is always empty. (This does raise a question of -- what if there are multiple accounturi parameters, but they all have identical values?)

But the intersection of two parameter tags with values "DNS-01,HTTP-01" and "DNS-01" is non-empty and eminently satisfiable.

I think it's okay to say that we don't want to bear the complexity burden of doing that kind of set intersection. But I think we should say that explicitly, since there is an obvious alternative.

Reply:
> This does raise a question of -- what if there are multiple accounturi parameters, but they all have identical values?

That's easy: the RFC specifically says that it is unsatisfiable.

> But the intersection of two parameter tags with values "DNS-01,HTTP-01" and "DNS-01" is non-empty and eminently satisfiable.

That may be true, but `validationmethods = dns-01 ; validationmethods = dns-01,http-01 ` is certainly a misconfiguration by the user because it would more succinctly be expressed as simply `validationmethods = dns-01`.